File: src/main/java/com/sup/dev/java_pc/sql/ResultRows.java
package com.sup.dev.java_pc.sql;


import com.sup.dev.java.classes.collections.AnyArray;
import com.sup.dev.java.libs.debug.Debug;

public class ResultRows {

    public final int rowsCount;
    public final int fieldsCount;
    public AnyArray values;

    public ResultRows(int rowsCount, AnyArray values) {
        this.rowsCount = rowsCount;
        this.fieldsCount = rowsCount == 0 ? 0 : (values.size() / rowsCount);
        this.values = values;
    }

    public boolean isEmpty() {
        return values.isEmpty();
    }

    public String toString(){
        StringBuilder s = new StringBuilder("ResultRows:");
        for (Object o : values.getList())
            s.append(" ").append(o);
        return s.toString();
    }

    public <K> K next(){
         K k = values.next();
        return k;
    }

    public Long[] asLongArray(){
        Long[] a = new Long[rowsCount];
        for(int i =0; i < rowsCount; i++) a[i] = next();
        return a;
    }

}

File: src/main/java/com/sup/dev/java_pc/sql/ResultRows.kt
package com.sup.dev.java_pc.sql

import com.sup.dev.java.classes.collections.AnyArray


class ResultRows(val rowsCount: Int, var values: AnyArray) {
    val fieldsCount: Int

    val isEmpty: Boolean
        get() = values.isEmpty()

    init {
        this.fieldsCount = if (rowsCount == 0) 0 else values.size() / rowsCount
    }

    override fun toString(): String {
        val s = StringBuilder("ResultRows:")
        for (o in values.list)
            s.append(" ").append(o)
        return s.toString()
    }

    operator fun <K> next(): K? {
        return values.next<K>()
    }

    operator fun hasNext(): Boolean {
        return values.hasNext()
    }

    fun longOrZero(): Long {
        return if (hasNext()) next()!!
        else 0L
    }

    fun sumOrZero(): Long {
        return if (hasNext()) Sql.parseSum(next<Any>())
        else 0L
    }

}
